spec/requests/api/docs/v3/statements_spec.rb

```diff
@@ -5,81 +5,18 @@
 
   let(:statement) { FactoryBot.create(:statement, active_lead_provider:) }
 
-  path "/api/v3/statements" do
-    get "Retrieve financial statements" do
-      tags "Statements"
-      produces "application/json"
-      security [api_key: []]
-
-      parameter name: :filter,
-                in: :query,
-                required: false,
-                schema: {
-                  "$ref": "#/components/schemas/StatementsFilter",
-                },
-                style: "deepObject"
-
-      parameter name: :page,
-                in: :query,
-                required: false,
-                schema: {
-                  "$ref": "#/components/schemas/PaginationFilter",
-                },
-                style: "deepObject"
-
-      response "200", "A list of statements as part of which the DfE will make output payments for participants" do
-        schema({ "$ref": "#/components/schemas/StatementsResponse" })
-
-        run_test!
-      end
-
-      response "401", "Unauthorized" do
-        let(:token) { "invalid" }
-
-        schema({ "$ref": "#/components/schemas/UnauthorisedResponse" })
-
-        run_test!
-      end
-    end
-  end
-
-  path "/api/v3/statements/{id}" do
-    get "Retrieve a specific financial statement" do
-      tags "Statements"
-      produces "application/json"
-      security [api_key: []]
-
-      parameter name: :id,
-                in: :path,
-                required: true,
-                schema: {
-                  "$ref": "#/components/schemas/IDAttribute",
-                }
-
-      response "200", "A specific financial statement" do
-        let(:id) { statement.api_id }
-
-        schema({ "$ref": "#/components/schemas/StatementResponse" })
-
-        run_test!
-      end
-
-      response "404", "Not found" do
-        let(:id) { SecureRandom.uuid }
-
-        schema({ "$ref": "#/components/schemas/NotFoundResponse" })
-
-        run_test!
-      end
-
-      response "401", "Unauthorized" do
-        let(:id) { statement.api_id }
-        let(:token) { "invalid" }
-
-        schema({ "$ref": "#/components/schemas/UnauthorisedResponse" })
-
-        run_test!
-      end
-    end
+  it_behaves_like "an API index endpoint documentation",
```

Comment:
Im not sure I agree with moving this to a shared context, it makes trying to understand the swagger docs quite hard, where before it was quite easy to follow. Happy to hear other thoughts, @ethax-ross what do you think?

Reply:
I don't feel too strongly either way; I think I introduced this pattern in NPQ reg for documentation requests - similar to the API request specs there's a lot of boilerplate when we start adding more endpoints and I think having it in a shared context cleans it up. It could probably be improved though, I think you can use keyword arguments so its clearer what we're passing into the shared contextr

Reply:
ok that's fair, happy to go with shared contexts if we add keyword arguments for clarity, thanks both :)